File: backend/app/schemas/evidence_score.py
"""Pydantic v2 schemas for the Evidence Score."""

from pydantic import BaseModel, ConfigDict


class EvidenceScoreBreakdown(BaseModel):
    """Weighted contribution of each dimension."""

    trust_weighted: float
    completeness_weighted: float
    freshness_weighted: float
    gap_penalty_weighted: float

    model_config = ConfigDict(from_attributes=True)


class EvidenceScoreRead(BaseModel):
    """Unified evidence score (0-100) for a building."""

    building_id: str
    score: int
    grade: str  # A, B, C, D, F
    trust: float
    completeness: float
    freshness: float
    gap_penalty: float
    breakdown: EvidenceScoreBreakdown
    computed_at: str

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/api/evidenceScore.ts
import { apiClient } from '@/api/client';

export interface EvidenceScoreBreakdown {
  trust_weighted: number;
  completeness_weighted: number;
  freshness_weighted: number;
  gap_penalty_weighted: number;
}

export interface EvidenceScore {
  building_id: string;
  score: number;
  grade: string;
  trust: number;
  completeness: number;
  freshness: number;
  gap_penalty: number;
  breakdown: EvidenceScoreBreakdown;
  computed_at: string;
}

export const evidenceScoreApi = {
  getEvidenceScore: async (buildingId: string): Promise<EvidenceScore> => {
    const response = await apiClient.get<EvidenceScore>(`/buildings/${buildingId}/evidence-score`);
    return response.data;
  },
  getPortfolioEvidenceScores: async (): Promise<EvidenceScore[]> => {
    const response = await apiClient.get<EvidenceScore[]>('/portfolio/evidence-scores');
    return response.data;
  },
};
